ckulkar/mocha17-dev
# Issue #6, 9 and 20: Changes for main screen UX and functionality

**Changes done:**
1. Main screen UX using Material design.
2. SettingsFragment for displaying the user-configurable options. Based on the state of a 'global' setting, all the other settings are enabled or disabled.
3. Status text displayed in the ToolBar that describes the current settings in human-readable form. The status text auto-updates on preference change, so do relevant preference summaries.
4. 'Select apps' feature: a list of apps installed on the device is displayed in a Dialog using a RecyclerView. The list is sorted by selection state (with selected at top) first and then by name. This provides an easy way to access selected apps at top of the list. 
The dialog also has an 'all apps' toggle to select all of the apps.
5. Other options like turning on/off the Android Wear requirement etc.
6. The preferences are stored using standard mechanisms and are accessible throughout the app.
7. Some refactoring, including moving classes to packages that make sense.

**Testing done:**
1. Launch app, verify that Setup page displays text and spinner correctly. This was necessary because the app theme has now changed.
2. Verify that MainActivity is launched on successful setup.
3. Verify that toggled settings result in correct text in status area.
4. Verify that settings are persisted across app launches.
5. Verify that 'select apps' dialog works as expected - displays list of apps, shows selected at top, hides all when 'all apps' is selected.
6. 

**Known issues:**
1. Orientation changes are not addressed in 'Select apps' dialog, sometimes resulting in a crash. This should be fixed soon.

Also note that #20 talks about scrolling away part of the ToolBar. That would be done later, a separate issue is being logged for tracking
